File: rb/features.rb
# NidApplicationSystem SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NidApplicationSystemFeatures
  def self.make_feature(name)
    case name
    when "base"
      NidApplicationSystemBaseFeature.new
    when "test"
      NidApplicationSystemTestFeature.new
    else
      NidApplicationSystemBaseFeature.new
    end
  end
end

File: py/features.py
# NidApplicationSystem SDK feature factory

from feature.base_feature import NidApplicationSystemBaseFeature
from feature.test_feature import NidApplicationSystemTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NidApplicationSystemBaseFeature(),
        "test": lambda: NidApplicationSystemTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
